Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -213,14 +213,19 @@ private class TopSitesCollectionView: UICollectionView {
 
 private class TopSitesLayout: UICollectionViewLayout {
     private var thumbnailRows: Int {
-        return max(2, Int((self.collectionView?.frame.height ?? self.thumbnailHeight) / self.thumbnailHeight))
+        return max(2, Int((self.height ?? self.thumbnailHeight) / self.thumbnailHeight))
     }
 
     private var thumbnailCols = 2
     private var thumbnailCount: Int {
         return thumbnailRows * thumbnailCols
     }
-    private var width: CGFloat { return self.collectionView?.frame.width ?? 0 }
+
+    private lazy var width: CGFloat = { return self.collectionView?.frame.width ?? 0 }()
+
+    private lazy var height: CGFloat? = {
+        return self.collectionView?.frame.height
+        }()
```

Comment:
Actually could we make height non-optional and do:
```
return self.collectionView?.frame.height ?? self.thumbnail height
```

So we don't need to do the optional check on line 216?